public/components/wz-filter-bar/wz-filter-bar.js

```diff
@@ -14,292 +14,302 @@ import { EuiComboBox } from '@elastic/eui';
+      // Select the option.
+      this.setState(
+        (prevState) => ({
+          selectedOptions: prevState.selectedOptions.concat(newOption),
+        }),
+        () => {
+          if (searchValue.includes(':')) {
+            this.onChange(this.state.selectedOptions);
+          }
         }
-      }
-    );
-    this.setState({ isProcessing: true });
-  };
+      );
+      this.setState({ isProcessing: true });
+    };
 
-  buildQuery = () => {
-    try {
-      const selectedOptions = this.state.selectedOptions;
-      const queryObj = {
-        query: '',
-        search: ''
-      };
-      const queryElements = selectedOptions.filter(x => {
-        return x.type !== 'search';
-      });
-      const searchElement = selectedOptions.filter(x => {
-        return x.type === 'search';
-      });
-      if (searchElement.length) {
-        queryObj.search = searchElement[0].label;
-      }
-      const twoOrMoreElements = queryElements.length > 1;
-      if (twoOrMoreElements) {
-        queryObj.query += '(';
-      }
-      queryElements.forEach((option, idx) => {
-        if (idx != 0) {
-          queryObj.query += option.type === 'AND' ? ';' : ',';
+    buildQuery = () => {
+      try {
+        const selectedOptions = this.state.selectedOptions;
+        const queryObj = {
+          query: '',
+          search: '',
+        };
+        const queryElements = selectedOptions.filter((x) => {
+          return x.type !== 'search';
+        });
+        const searchElement = selectedOptions.filter((x) => {
+          return x.type === 'search';
+        });
+        if (searchElement.length) {
+          queryObj.search = searchElement[0].label;
         }
-        queryObj.query += option.query
-          ? option.query
-          : option.label.replace(':', '=');
-      });
-      if (twoOrMoreElements) {
-        queryObj.query += ')';
-      }
-      this.props.clickAction({
-        q: queryObj.query,
-        search: queryObj.search,
-        selectedOptions: selectedOptions
-      });
-    } catch (error) {} // eslint-disable-line
-  };
+        const twoOrMoreElements = queryElements.length > 1;
+        if (twoOrMoreElements) {
+          queryObj.query += '(';
+        }
+        queryElements.forEach((option, idx) => {
+          if (idx != 0) {
+            queryObj.query += option.type === 'AND' ? ';' : ',';
+          }
+          queryObj.query += option.query ? option.query : option.label.replace(':', '=');
+        });
+        if (twoOrMoreElements) {
+          queryObj.query += ')';
+        }
+        this.props.clickAction({
+          q: queryObj.query,
+          search: queryObj.search,
+          selectedOptions: selectedOptions,
+        });
+      } catch (error) {
+        const options = {
+          context: `${WzFilterBar.name}.buildQuery`,
+          level: UI_LOGGER_LEVELS.ERROR,
+          severity: UI_ERROR_SEVERITIES.BUSINESS,
+          store: true,
```

Comment:
always when a query build fails will it be stored? can it be caused by a user intentionally?